File: programmers/43162.py
# 출처 : https://programmers.co.kr/learn/courses/30/lessons/43162

from collections import deque


def solution(n, computers):
    global answer
    answer = 0
    visit = []
    for i in range(n):
        visit.append(False)

    def bfs(i):
        if(visit[i]):
            return

        global answer
        answer += 1
        queue = deque([i])

        while(queue):
            nowNode = queue.popleft()
            visit[nowNode] = True

            for i in range(n):
                if(computers[nowNode][i] == 1 and not visit[i]):
                    queue.append(i)

    for i in range(n):
        bfs(i)

    return answer

File: programmers/43162.js

function solution(n, computers) {
    let answer = 0;
    let visit = []

    for (let i = 0; i < n; i++) {
        visit.push(0)
    }

    for (let i = 0; i < n; i++) {
        if (visit[i] == 0) {
            dfs(i);
            answer += 1
        }
    }
    function dfs(x) {
        visit[x] = 1;
        for (let i = 0; i < n; i++) {
            if (computers[x][i] == 1 && visit[i] == 0) {
                dfs(i);
            }
        }
    }

    // console.log(answer);
    return answer;
}

solution(3, [[1, 1, 0], [1, 1, 0], [0, 0, 1]])